ldap/client/api/src/main/java/org/apache/directory/ldap/client/api/LdapNetworkConnection.java

```diff
@@ -235,6 +238,12 @@
     /** The Ldap codec protocol filter */
     private IoFilter ldapProtocolFilter = new ProtocolCodecFilter( codec.getProtocolCodecFactory() );
 
+    /** The cached thread pool used when trying to find a workable LDAP connection. */
+    private static ExecutorService executor = Executors.newCachedThreadPool();
+
+    /** The completion service helps us out when we need to wait for the LDAP connection futures to finish executing. */
+    private CompletionService<LdapNetworkConnectionFuture> completionService;
```

Comment:
does this need to be a class member? Can't it be created in and destroyed in the method?